Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/ProgressIndicators/ProgressIndicatorObjectDisplay.cs

```diff
@@ -0,0 +1,136 @@
+
+        [Header("How fast does object rotate?")]
+        [SerializeField]
+        private float rotationIncrement = 200;
+
+        [Header("Start scale of the object?")]
+        [SerializeField]
+        private float minScale = 1.0f;
+
+        [Header("Final scale of the object?")]
+        [SerializeField]
+        private float maxScale = 9.0f;
+
+        [Header("Should object rotate?")]
+        [SerializeField]
+        private bool rotationActive = false;
+
+        [Header("Rotation occurs about which axes?")]
+        [SerializeField]
+        private bool xAxisRotation = false;
+        [SerializeField]
+        private bool yAxisRotation = true;
+        [SerializeField]
+        private bool zAxisRotation = false;
+
+        [Header("Grow / Shrink curve on open / close")]
+        [SerializeField]
+        [Tooltip("Curve for opening animation. MUST end with value >= 1 or transition will not complete.")]
+        private AnimationCurve openCurve = AnimationCurve.EaseInOut(0, 0, 1, 1);
+        [SerializeField]
+        [Tooltip("Curve for opening animation. MUST end with value <= 0 or transition will not complete.")]
+        private AnimationCurve closeCurve = AnimationCurve.EaseInOut(0, 1, 1, 0);
+
+        [SerializeField]
+        private TextMeshPro messageText = null;
+
+        [SerializeField]
+        private ProgressIndicatorState state = ProgressIndicatorState.Closed;
+
+        [SerializeField]
+        [Range(0f, 1f)]
+        private float progress;
+
+        private float currentScale;
+        private float elapsedTime;
+
+        public async Task OpenAsync()
+        {
+            gameObject.SetActive(true);
+
+            Reset();
+
+            state = ProgressIndicatorState.Opening;
+
+            float startTime = Time.unscaledTime;
+            float openScale = 0f;
+            while (openScale < 1)
+            {
+                openScale = openCurve.Evaluate(Time.unscaledTime - startTime);
+                scaleTargetObject.transform.localScale = Vector3.one * currentScale * openScale;
+                await Task.Yield();
+            }
+
+            state = ProgressIndicatorState.Open;
+        }
+
+        public async Task CloseAsync()
+        {
+            state = ProgressIndicatorState.Closing;
+
+            float startTime = Time.unscaledTime;
+            float closeScale = 1f;
+            while (closeScale > 0)
+            {
+                closeScale = closeCurve.Evaluate(Time.unscaledTime - startTime);
+                scaleTargetObject.transform.localScale = Vector3.one * currentScale * closeScale;
+                await Task.Yield();
+            }
+
+            state = ProgressIndicatorState.Closed;
+
+            gameObject.SetActive(false);
+        }
+
+        private void Reset()
+        {
+            elapsedTime = 0.0f;
+            currentScale = minScale;
+        }
+
+        private void Update()
+        {
+            elapsedTime += Time.unscaledDeltaTime;
```

Comment:
Is elapsedTime still being used for something in this class? It looks like it gets set, and then updated, but never used for any other interesting part later in this file.